test/tabulated_tests.jl

```diff
@@ -0,0 +1,225 @@
+fail immediately due to a non-existent file.
+
+Checking for duplicate files can be helpful
+by avoiding unexpected duplicate work.
+"""
+function validate_tests(unit_tests::Vector{UnitTest}; test_path)
+    # Test uniqueness of included files
+    err = :pass
+    filenames = map(x -> x.filename, unit_tests)
+    if !allunique(filenames) # let's not do things more than once
+        for (key, val) in countmap(filenames)
+            val > 1 || continue
+            @info "Duplicate file found: $key, ($val times)"
+            err = :duplicate_file
+        end
+    end
+    # Test that files exist
+    if !all(map(x -> isfile(x.filename), unit_tests))
+        for filename in filenames
+            rfile = joinpath(test_path, filename)
+            isfile(rfile) && continue
+            @warn "Filename: $rfile does not exist"
+            err = :non_existent_file
+        end
+    end
+    return err
+end
+
+
+"""
+    tabulate_tests(
+        unit_tests::Vector{UnitTest};
+        include_timings::Bool = true,
+        time_format::Symbol = :second, # one of (:second, :compoundperiod)
+    )
+
+ - `include_timings::Bool` indicates whether or not to include the timings/percent columns
+ - `time_format::Symbol` specify the time format. Valid values include [:second (default), :compoundperiod].
+                         If we do not match either of these, then we warn & default to `:second`.
+
+Tabulate the given unit tests. The `include_timings` kwarg
+extends the table to include timings of the tests.
+"""
+function tabulate_tests(
+    unit_tests::Vector{UnitTest};
+    include_timings::Bool = true,
+    time_format::Symbol = :second,
+)
+    title =
+        include_timings ? "Tests results" :
+        "Running the following unit tests..."
+    if include_timings
+        sort!(unit_tests; by = x -> x.elapsed)
+        local time_header
+        elapsed_times = if time_format == :compoundperiod
+            time_header = "Time"
+            map(x -> time_and_units_str(x.elapsed), unit_tests)
+        elseif time_format == :second
+            time_header = "Time (s)"
+            map(x -> x.elapsed, unit_tests)
+        else
+            @warn "Invalid time format `$time_format`. Falling back on `:second`"
+            time_header = "Time (s)"
+            map(x -> x.elapsed, unit_tests)
+        end
+        ∑times = sum(x -> x.elapsed, unit_tests)
+        time_percent = map(x -> x.elapsed / ∑times * 100, unit_tests)
+        header = ["% Time", time_header, "Name", "Filename"]
+        data = hcat(
+            time_percent,
+            elapsed_times,
+            map(x -> x.name, unit_tests),
+            map(x -> x.filename, unit_tests),
+        )
+    else
+        header = ["Name", "Filename"]
+        data =
+            hcat(map(x -> x.name, unit_tests), map(x -> x.filename, unit_tests))
+    end
+    PrettyTables.pretty_table(
+        data;
+        title,
+        header,
+        alignment = :l,
+        crop = :horizontal,
```

Comment:
Does the table need to be cropped horizontally?

Reply:
It's pretty jumbled on GHA when we don't. We could change the behavior in CI vs interactive, but the last column is the filename and the test name usually fits on the page. I mostly put `crop = :horizontal` to ensure that it's not cropped in the vertical (I think the default is `crop = :both`).